presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/SimpleNodeSelector.java

```diff
@@ -149,14 +151,21 @@ public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTas
                 throw new PrestoException(NO_NODES_AVAILABLE, "No nodes available to run query");
             }
 
-            InternalNode chosenNode = chooseLeastBusyNode(candidateNodes, assignmentStats::getTotalSplitCount, preferredNodeCount, maxSplitsPerNode);
-            if (chosenNode == null) {
+            NodeResult chosenNode = chooseLeastBusyNode(candidateNodes, assignmentStats::getTotalSplitCount, preferredNodeCount, maxSplitsPerNode);
+            if (!chosenNode.getNode().isPresent()) {
```

Comment:
How about we keep `chosenNode` to null for this case? I don't know the history of why this is `null` rather than `Optional.empty()`, but I personally would prefer the latter..

Reply:
Do you want to keep `chosenNode.getNode() ` `null` or `Optional.empty()`?

Also the reason we create data structure `NodeResult` is to provide the ability to represent following situation:
chosenNode is present, chosenNode should be cached
chosenNode is present, chosenNode should not be cached
chosenNode is not present

Reply:
I don't have strong opinion around `null` vs `empty`, but I'd prefer we make decision based on `chosenNode` being null/empty, instead of `chosenNode.getNode` :)